kitsune/questions/static/questions/js/components/UserAuth.jsx

```diff
@@ -1,97 +1,49 @@
 /* globals React:false */
 import AAQStep from './AAQStep.jsx';
 import UserAuthActions from '../../../users/js/actions/UserAuthActions.es6.js';
+import {authStates} from '../../../users/js/constants/UserAuthConstants.es6.js';
+import LoginForm from '../../../users/js/components/LoginForm.jsx';
+import RegisterForm from '../../../users/js/components/RegisterForm.jsx';
 
 export default class UserAuth extends AAQStep {
   constructor(props) {
     super(props);
-    this.state = {
-      signingIn: false,
-    };
+  }
+
+  componentDidMount() {
+    UserAuthActions.checkAuthState()
+    .then(authState => {
+      if (authState === authStates.LOGGED_OUT) {
+        UserAuthActions.showRegister();
+      }
+    });
   }
 
   heading() {
-    if (this.props.loggedIn) {
-      return `Signed in as ${this.props.username}`;
-    } else {
-      return 'Almost there! We just need to create a support forum account!';
+    const messages = {
+      [authStates.LOGGED_IN]: `Signed in as ${this.props.userAuth.username}`,
+      [authStates.REGISTERING]: 'Almost there! We just need to create a support forum account!',
+      [authStates.LOGGING_IN]: 'Log in to an existing account',
+    };
+
+    let message = messages[this.props.userAuth.state] || '';
+    if (this.props.userAuth.pending) {
+      message += ' (pending)';
     }
+
+    return message;
   }
 
   body() {
-    if (this.props.loggedIn) {
-      return null;
-    }
-    if (this.state.signingIn) {
-      // return <LoginForm/>;
+    if (this.props.userAuth.state === authStates.REGISTERING) {
+      return <RegisterForm userAuth={this.props.userAuth}/>;
+    } else if (this.props.userAuth.state === authStates.LOGGING_IN) {
+      return <LoginForm userAuth={this.props.userAuth}/>;
     } else {
-      return <RegisterForm/>;
+      return null;
     }
   }
 }
 UserAuth.propTypes = {
-  loggedIn: React.PropTypes.bool.isRequired,
+  userAuth: React.PropTypes.object.isRequired,
 };
-
-class RegisterForm extends React.Component {
```

Comment:
This form moved, largely unchanged, to `kitsune/users/static/users/js/components/RegisterForm.jsx`.